frontend/packages/dev-console/integration-tests/views/git-import-flow.view.ts

```diff
@@ -99,6 +228,82 @@ export const addApplicationWithExistingApps = async function(name: string, nodeN
   await safeSendKeys(appName, 'appName', nodeName);
 };
 
+export const addApplicationInGeneral = async function(name: string, nodeName: string) {
+  browser.wait(until.visibilityOf(await element(by.css('[id$=application-name-field]'))), waitForElement);
+    await element(by.css('[id$=application-name-field]')).getTagName()
+    .then(async(tagName) => {
+      if(tagName.includes('button')) {
+        await addApplicationWithExistingApps(name, nodeName);
+      }  else {
+        await addApplication(name, nodeName);
+      }          
+    })   
+}
+
 export const setBuilderImage = async function() {
   await builderImage.click();
 };
+
+// Automating Advanced options present in git import flow
+export const setRouting = async function(hostname:string, path: string) {
+  await enterText(routingObj.hostname, hostname);
+  await enterText(routingObj.path, path);
+  await selectByIndex(routingObj.targetPort);
+};
+
+export const setSecureRoute = async function(tlsTerminationValue: TLSTerminationValues, insecureTrafficValue: string = 'None') {
+  // if(tlsTerminationValue ==TLSTerminationValues.Edge || tlsTerminationValue ==TLSTerminationValues.ReEncrypt) {
+    await click(routingObj.secureRoute).then(async() => {
+    await browser.wait(until.elementToBeClickable(routingObj.tlsTermination), waitForElement, `Unable to view the TLS termination dropdown field, even after ${waitForElement} ms `)
+    await selectByVisibleText(routingObj.tlsTermination, tlsTerminationValue);
+    // await routingObj().certificate.sendKeys('a');
+    // await routingObj().privateKey.sendKeys();
+    // await routingObj().caCertificate.sendKeys();
+    await selectByVisibleText(routingObj.insecureTraffic, insecureTrafficValue);
+    });
+  // } else if(tlsTerminationValue ==TLSTerminationValues.Passthrough) {
+  //   await click(routingObj().secureRoute).then(async() => {
+  //     await browser.wait(until.elementToBeClickable(routingObj().tlsTermination), waitForElement, `Unable to view the TLS termination dropdown field, even after ${waitForElement} ms `)
+  //     await selectByVisibleText(routingObj().tlsTermination, tlsTerminationValue);
+  //     await selectByVisibleText(routingObj().insecureTraffic, insecureTrafficValue);
+  //   });
+  // } else {
+  //   throw new Error('Some thing went wrong');
+  // }
```

Comment:
Commented code, remove.
```suggestion
    await click(routingObj.secureRoute).then(async() => {
    await browser.wait(until.elementToBeClickable(routingObj.tlsTermination), waitForElement, `Unable to view the TLS termination dropdown field, even after ${waitForElement} ms `)
    await selectByVisibleText(routingObj.tlsTermination, tlsTerminationValue);
    await selectByVisibleText(routingObj.insecureTraffic, insecureTrafficValue);
    });
```

Reply:
Done